File: python/38b_1857A_ArrayColoring.py
def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        arr = map(int, input().split())
        no_of_even, no_of_odd = 0, 0
        for i in arr:
            if i % 2 == 0:
                no_of_even += 1
            else:
                no_of_odd += 1
        if no_of_odd % 2 == 0:
            ans.append("YES")
        else:
            ans.append("NO")
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()

File: cpp/38b_1857A_ArrayColoring.cpp
#include<bits/stdc++.h>

int main(){
    int t, n, temp, no_of_even, no_of_odd;
    std::vector<std::string> ans;
    std::cin>>t;
    for(int i = 0; i < t; i++){
        std::cin>>n;
        no_of_even = 0, no_of_odd = 0;
        for(int j = 0; j < n; j++){
            std::cin>>temp;
            if (temp % 2 == 0){
                no_of_even++;
            }
            else{
                no_of_odd++;
            }
        }
        if(no_of_odd % 2 == 0){
            ans.push_back("YES");
        }
        else{
            ans.push_back("NO");
        }
    }
    for (auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}